Doc/library/typing.rst

```diff
@@ -428,12 +428,12 @@ value of type :data:`Any` and assign it to any variable::
 
    from typing import Any
 
-   a = None    # type: Any
-   a = []      # OK
-   a = 2       # OK
+   a: Any = None
+   a = []          # OK
+   a = 2           # OK
 
-   s = ''      # type: str
-   s = a       # OK
+   s = ''          # the type-checker infers the type of s is str
```

Comment:
Sorry, I'm not a native speaker, but for me `infers the type of s as str` looks more natural.

@willingc could you take a quick look at the wording?

Reply:
Hmm, I prefer "is".

How about `the type-checker understands the type of s is str`?

Reply:
How about `#Inferred type of s is str` ? Seems like it's how PEP 484 describes things https://www.python.org/dev/peps/pep-0484/#instantiating-generic-classes-and-type-erasure.

Personally, I think both `as` and `is` are fine, but I'm not an English expert :).

Reply:
I like `#Inferred type of s is str`. Also, it is significantly shorter.

@AlexWaygood what do you think?

Reply:
My only quibble with `#Inferred type of s is str` is that it's not clear _who_ is doing the inferring 😆 but let's go with it, I agree that it's nicer to have something more concise 😀